### ISSUE TITLE
DynamoDB index query function has a wrong parameter sequence.

### ISSUE THREAD

**Issue by roselone** [aws/aws-sdk-java#2512]
In `QueryImpl.java`, the query method ask **filterExpression** before **projectionExpression**.
```JAVA
public ItemCollection<QueryOutcome> query(KeyAttribute hashKey,
            RangeKeyCondition rangeKeyCondition, String filterExpression,
            String projectionExpression, Map<String, String> nameMap,
            Map<String, Object> valueMap) { ... }
```

However, in `Index.java`, will pass **projectionExpression** before **filterExpression**
```JAVA
    @Beta
    public ItemCollection<QueryOutcome> query(KeyAttribute hashKey,
            RangeKeyCondition rangeKeyCondition, QueryExpressionSpec queryExpressions) {
        return queryDelegate.query(hashKey, rangeKeyCondition,
                queryExpressions.getProjectionExpression(),
                queryExpressions.getFilterExpression(),
                queryExpressions.getNameMap(), queryExpressions.getValueMap());
    
```

**Comment by debora-ito** [comment]
@roselone thank you for reporting this. Will take a look.

**Comment by debora-ito** [comment]
Worked on a repro case (just to be sure, even thought the error was pretty obvious), can confirm it as a bug.

**Comment by debora-ito** [comment]
@roselone a fix was release in version [`1.12.59`](https://github.com/aws/aws-sdk-java/blob/master/CHANGELOG.md#11259-2021-08-31). 

I'll go ahead and close this. Thank you again!
